MonkeyLoader.Resonite.Integration/Locale/ModLocaleDriverFix.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Elements.Core;
+using FrooxEngine;
+using HarmonyLib;
+using MonkeyLoader.Patching;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using EngineLocaleHelper = FrooxEngine.LocaleHelper;
+
+namespace MonkeyLoader.Resonite.Locale
+{
+    [HarmonyPatch(typeof(EngineLocaleHelper))]
+    [HarmonyPatchCategory(nameof(ModLocaleDriverFix))]
+    internal sealed class ModLocaleDriverFix : ResoniteMonkey<ModLocaleDriverFix>
+    {
+        protected override IEnumerable<IFeaturePatch> GetFeaturePatches() => Enumerable.Empty<IFeaturePatch>();
+
+        private static void AddFallbackMessage(IField<string> field, LocaleStringDriver localeDriver, string notFoundMessage, string fallbackMessage)
+        {
+            var booleanDriver = localeDriver.Slot.AttachComponent<BooleanValueDriver<string>>();
+            localeDriver.Target.Target = booleanDriver.FalseValue;
+            booleanDriver.TrueValue.Value = fallbackMessage;
+            booleanDriver.TargetField.Target = field;
+
+            var equalityDriver = localeDriver.Slot.AttachComponent<ValueEqualityDriver<string>>();
+            equalityDriver.Reference.Value = notFoundMessage;
+            equalityDriver.TargetValue.Target = booleanDriver.FalseValue;
+            equalityDriver.Target.Target = booleanDriver.State;
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch(nameof(EngineLocaleHelper.DriveLocalized), [typeof(IField<string>), typeof(string), typeof(string), typeof(Dictionary<string, object>)])]
+        private static void DriveLocalizedPostfix(IField<string> field, string key, string? format, Dictionary<string, object>? arguments, LocaleStringDriver __result)
```

Comment:
Do you need to patch these? Adding too many patches creates more points of failure that could break if there is an update to Resonite.

Reply:
It's the easiest place to implement it in while ensuring that mod locale keys get treated correctly (i.e. have the fallback added). Doing it there ensures that it'll work regardless of how exactly the mod locale keys get assigned through built-in systems, as they're the methods used by all the others.

With the encapsulation and being postfixes only, any failures wouldn't matter for the main functionality of the methods either, so I'd not be worried about that.

Could actually rather call it ModLocaleDriverFallback or something instead of ...Fix.